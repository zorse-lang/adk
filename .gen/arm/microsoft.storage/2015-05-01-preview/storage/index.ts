import { ArmResource, Entity as ADKEntity } from "@zorse/adk";

export class storageAccounts
	extends ArmResource<storageAccountsComponentInputs>
	implements storageAccountsComponentOutputs
{
	constructor(entity: ADKEntity, options: storageAccountsComponentInputs) {
		super(entity, options.name, "Microsoft.Storage/storageAccounts", "2015-05-01-preview", options);
	}
	public readonly apiVersion: "2015-05-01-preview";
	public readonly id: string;
	public readonly type: "Microsoft.Storage/storageAccounts";
}
export interface storageAccountsComponentOutputs {
	readonly apiVersion: "2015-05-01-preview";
	readonly id: string;
	readonly type: "Microsoft.Storage/storageAccounts";
}
export interface storageAccountsComponentInputs {
	readonly location: string;
	readonly name: string;
	readonly properties?: StorageAccountPropertiesCreateParametersOrStorageAccountProperties | undefined;
	readonly tags?: ResourceTags | undefined;
}
export function listKeys(resource: storageAccounts): StorageAccountKeys {
	if (resource.apiVersion !== "2015-05-01-preview") {
		throw new Error(`listKeys is not available for api version ${resource.apiVersion}`);
	}
	if (resource.type !== "Microsoft.Storage/storageAccounts") {
		throw new Error(`listKeys is not available for resource type ${resource.type}`);
	}
	throw new Error("not implemented");
}
export interface CustomDomain {
	readonly name?: string | undefined;
	readonly useSubDomainName?: boolean | undefined;
}
export interface Endpoints {
	readonly blob?: string | undefined;
	readonly queue?: string | undefined;
	readonly table?: string | undefined;
}
export interface ResourceTags {
	readonly "[ key: string ]"?: string | undefined;
}
export interface StorageAccountKeys {
	readonly key1?: string | undefined;
	readonly key2?: string | undefined;
}
export interface StorageAccountPropertiesCreateParametersOrStorageAccountProperties {
	readonly accountType?: ("Premium_LRS" | "Standard_GRS" | "Standard_LRS" | "Standard_RAGRS") | undefined;
	readonly creationTime?: string | undefined;
	readonly customDomain?: CustomDomain | undefined;
	readonly lastGeoFailoverTime?: string | undefined;
	readonly primaryEndpoints?: Endpoints | undefined;
	readonly primaryLocation?: string | undefined;
	readonly provisioningState?: ("Creating" | "ResolvingDNS") | undefined;
	readonly secondaryEndpoints?: Endpoints | undefined;
	readonly secondaryLocation?: string | undefined;
	readonly statusOfPrimary?: "Available" | undefined;
	readonly statusOfSecondary?: "Available" | undefined;
}
export default {
	storageAccounts: storageAccounts,
};
