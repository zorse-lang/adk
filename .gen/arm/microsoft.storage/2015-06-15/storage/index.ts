import { ArmResource, Entity as ADKEntity } from "@zorse/adk";

export class storageAccounts
	extends ArmResource<storageAccountsComponentInputs>
	implements storageAccountsComponentOutputs
{
	constructor(entity: ADKEntity, options: storageAccountsComponentInputs) {
		super(entity, options.name, "Microsoft.Storage/storageAccounts", "2015-06-15", options);
	}
	public readonly apiVersion: "2015-06-15";
	public readonly id: string;
	public readonly type: "Microsoft.Storage/storageAccounts";
}
export interface storageAccountsComponentOutputs {
	readonly apiVersion: "2015-06-15";
	readonly id: string;
	readonly type: "Microsoft.Storage/storageAccounts";
}
export interface storageAccountsComponentInputs {
	readonly location: string;
	readonly name: string;
	readonly properties?: StorageAccountPropertiesCreateParametersOrStorageAccountProperties;
	readonly tags?: StorageAccountCreateParametersTags;
}
export function listKeys(resource: storageAccounts): StorageAccountKeys {
	if (resource.apiVersion !== "2015-06-15") {
		throw new Error(`listKeys is not available for api version ${resource.apiVersion}`);
	}
	if (resource.type !== "Microsoft.Storage/storageAccounts") {
		throw new Error(`listKeys is not available for resource type ${resource.type}`);
	}
	throw new Error("not implemented");
}
export interface CustomDomain {
	readonly name: string;
	readonly useSubDomainName?: boolean;
}
export interface Endpoints {
	readonly blob?: string;
	readonly file?: string;
	readonly queue?: string;
	readonly table?: string;
}
export interface StorageAccountCreateParametersTags {
	readonly [key: string]: string;
}
export interface StorageAccountKeys {
	readonly key1?: string;
	readonly key2?: string;
}
export interface StorageAccountPropertiesCreateParametersOrStorageAccountProperties {
	readonly accountType: "Premium_LRS" | "Standard_GRS" | "Standard_LRS" | "Standard_RAGRS";
	readonly creationTime?: string;
	readonly customDomain?: CustomDomain;
	readonly lastGeoFailoverTime?: string;
	readonly primaryEndpoints?: Endpoints;
	readonly primaryLocation?: string;
	readonly provisioningState?: "Creating" | "ResolvingDNS";
	readonly secondaryEndpoints?: Endpoints;
	readonly secondaryLocation?: string;
	readonly statusOfPrimary?: "Available";
	readonly statusOfSecondary?: "Available";
}
export default {
	storageAccounts: storageAccounts,
};
