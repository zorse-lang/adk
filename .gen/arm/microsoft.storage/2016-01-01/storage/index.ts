import { ArmResource, Entity as ADKEntity } from "@zorse/adk";

export class storageAccounts
	extends ArmResource<storageAccountsComponentInputs>
	implements storageAccountsComponentOutputs
{
	constructor(entity: ADKEntity, options: storageAccountsComponentInputs) {
		super(entity, options.name, "Microsoft.Storage/storageAccounts", "2016-01-01", options);
	}
	public readonly apiVersion: "2016-01-01";
	public readonly id: string;
	public readonly type: "Microsoft.Storage/storageAccounts";
}
export interface storageAccountsComponentOutputs {
	readonly apiVersion: "2016-01-01";
	readonly id: string;
	readonly type: "Microsoft.Storage/storageAccounts";
}
export interface storageAccountsComponentInputs {
	readonly kind: "BlobStorage";
	readonly location: string;
	readonly name: string;
	readonly properties?: StorageAccountPropertiesCreateParametersOrStorageAccountProperties | undefined;
	readonly sku: Sku;
	readonly tags?: StorageAccountCreateParametersTags | undefined;
}
export function listKeys(resource: storageAccounts): StorageAccountListKeysResult {
	if (resource.apiVersion !== "2016-01-01") {
		throw new Error(`listKeys is not available for api version ${resource.apiVersion}`);
	}
	if (resource.type !== "Microsoft.Storage/storageAccounts") {
		throw new Error(`listKeys is not available for resource type ${resource.type}`);
	}
	throw new Error("not implemented");
}
export interface CustomDomain {
	readonly name: string;
	readonly useSubDomainName?: boolean | undefined;
}
export interface Encryption {
	readonly keySource: "Microsoft.Storage";
	readonly services?: EncryptionServices | undefined;
}
export interface EncryptionService {
	readonly enabled?: boolean | undefined;
	readonly lastEnabledTime?: string | undefined;
}
export interface EncryptionServices {
	readonly blob?: EncryptionService | undefined;
}
export interface Endpoints {
	readonly blob?: string | undefined;
	readonly file?: string | undefined;
	readonly queue?: string | undefined;
	readonly table?: string | undefined;
}
export interface Sku {
	readonly name: "Premium_LRS" | "Standard_GRS" | "Standard_LRS" | "Standard_RAGRS";
	readonly tier?: "Premium" | undefined;
}
export interface StorageAccountCreateParametersTags {
	readonly "[ key: string ]"?: string | undefined;
}
export interface StorageAccountKey {
	readonly keyName?: string | undefined;
	readonly permissions?: "FULL" | undefined;
	readonly value?: string | undefined;
}
export interface StorageAccountListKeysResult {
	readonly keys?: StorageAccountKey[] | undefined;
}
export interface StorageAccountPropertiesCreateParametersOrStorageAccountProperties {
	readonly accessTier?: "Cool" | undefined;
	readonly creationTime?: string | undefined;
	readonly customDomain?: CustomDomain | undefined;
	readonly encryption?: Encryption | undefined;
	readonly lastGeoFailoverTime?: string | undefined;
	readonly primaryEndpoints?: Endpoints | undefined;
	readonly primaryLocation?: string | undefined;
	readonly provisioningState?: ("Creating" | "ResolvingDNS") | undefined;
	readonly secondaryEndpoints?: Endpoints | undefined;
	readonly secondaryLocation?: string | undefined;
	readonly statusOfPrimary?: "Available" | undefined;
	readonly statusOfSecondary?: "Available" | undefined;
}
export default {
	storageAccounts: storageAccounts,
};
