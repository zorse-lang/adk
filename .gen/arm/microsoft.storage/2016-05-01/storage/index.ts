import { ArmResource, Entity as ADKEntity } from "@zorse/adk";

export class storageAccounts
	extends ArmResource<storageAccountsComponentInputs>
	implements storageAccountsComponentOutputs
{
	constructor(entity: ADKEntity, options: storageAccountsComponentInputs) {
		super(entity, options.name, "Microsoft.Storage/storageAccounts", "2016-05-01", options);
	}
	public readonly apiVersion: "2016-05-01";
	public readonly id: string;
	public readonly type: "Microsoft.Storage/storageAccounts";
}
export interface storageAccountsComponentOutputs {
	readonly apiVersion: "2016-05-01";
	readonly id: string;
	readonly type: "Microsoft.Storage/storageAccounts";
}
export interface storageAccountsComponentInputs {
	readonly kind: "BlobStorage";
	readonly location: string;
	readonly name: string;
	readonly properties?: StorageAccountPropertiesCreateParametersOrStorageAccountProperties | undefined;
	readonly sku: Sku;
	readonly tags?: StorageAccountCreateParametersTags | undefined;
}
export function listAccountSas(resource: storageAccounts, input: AccountSasParameters): ListAccountSasResponse {
	if (resource.apiVersion !== "2016-05-01") {
		throw new Error(`listAccountSas is not available for api version ${resource.apiVersion}`);
	}
	if (resource.type !== "Microsoft.Storage/storageAccounts") {
		throw new Error(`listAccountSas is not available for resource type ${resource.type}`);
	}
	throw new Error("not implemented");
}
export function listKeys(resource: storageAccounts): StorageAccountListKeysResult {
	if (resource.apiVersion !== "2016-05-01") {
		throw new Error(`listKeys is not available for api version ${resource.apiVersion}`);
	}
	if (resource.type !== "Microsoft.Storage/storageAccounts") {
		throw new Error(`listKeys is not available for resource type ${resource.type}`);
	}
	throw new Error("not implemented");
}
export function listServiceSas(resource: storageAccounts, input: ServiceSasParameters): ListServiceSasResponse {
	if (resource.apiVersion !== "2016-05-01") {
		throw new Error(`listServiceSas is not available for api version ${resource.apiVersion}`);
	}
	if (resource.type !== "Microsoft.Storage/storageAccounts") {
		throw new Error(`listServiceSas is not available for resource type ${resource.type}`);
	}
	throw new Error("not implemented");
}
export interface AccountSasParameters {
	readonly keyToSign?: string | undefined;
	readonly signedExpiry: string;
	readonly signedIp?: string | undefined;
	readonly signedPermission: "a" | "c" | "d" | "l" | "p" | "r" | "u" | "w";
	readonly signedProtocol?: "https" | undefined;
	readonly signedResourceTypes: "c" | "o" | "s";
	readonly signedServices: "b" | "f" | "q" | "t";
	readonly signedStart?: string | undefined;
}
export interface CustomDomain {
	readonly name: string;
	readonly useSubDomainName?: boolean | undefined;
}
export interface Encryption {
	readonly keySource: "Microsoft.Storage";
	readonly services?: EncryptionServices | undefined;
}
export interface EncryptionService {
	readonly enabled?: boolean | undefined;
	readonly lastEnabledTime?: string | undefined;
}
export interface EncryptionServices {
	readonly blob?: EncryptionService | undefined;
}
export interface Endpoints {
	readonly blob?: string | undefined;
	readonly file?: string | undefined;
	readonly queue?: string | undefined;
	readonly table?: string | undefined;
}
export interface ListAccountSasResponse {
	readonly accountSasToken?: string | undefined;
}
export interface ListServiceSasResponse {
	readonly serviceSasToken?: string | undefined;
}
export interface ServiceSasParameters {
	readonly canonicalizedResource: string;
	readonly endPk?: string | undefined;
	readonly endRk?: string | undefined;
	readonly keyToSign?: string | undefined;
	readonly rscc?: string | undefined;
	readonly rscd?: string | undefined;
	readonly rsce?: string | undefined;
	readonly rscl?: string | undefined;
	readonly rsct?: string | undefined;
	readonly signedExpiry?: string | undefined;
	readonly signedIdentifier?: string | undefined;
	readonly signedIp?: string | undefined;
	readonly signedPermission?: ("a" | "c" | "d" | "l" | "p" | "r" | "u" | "w") | undefined;
	readonly signedProtocol?: "https" | undefined;
	readonly signedResource: "b" | "c" | "f" | "s";
	readonly signedStart?: string | undefined;
	readonly startPk?: string | undefined;
	readonly startRk?: string | undefined;
}
export interface Sku {
	readonly name: "Premium_LRS" | "Standard_GRS" | "Standard_LRS" | "Standard_RAGRS";
	readonly tier?: "Premium" | undefined;
}
export interface StorageAccountCreateParametersTags {
	readonly "[ key: string ]"?: string | undefined;
}
export interface StorageAccountKey {
	readonly keyName?: string | undefined;
	readonly permissions?: "Full" | undefined;
	readonly value?: string | undefined;
}
export interface StorageAccountListKeysResult {
	readonly keys?: StorageAccountKey[] | undefined;
}
export interface StorageAccountPropertiesCreateParametersOrStorageAccountProperties {
	readonly accessTier?: "Cool" | undefined;
	readonly creationTime?: string | undefined;
	readonly customDomain?: CustomDomain | undefined;
	readonly encryption?: Encryption | undefined;
	readonly lastGeoFailoverTime?: string | undefined;
	readonly primaryEndpoints?: Endpoints | undefined;
	readonly primaryLocation?: string | undefined;
	readonly provisioningState?: ("Creating" | "ResolvingDNS") | undefined;
	readonly secondaryEndpoints?: Endpoints | undefined;
	readonly secondaryLocation?: string | undefined;
	readonly statusOfPrimary?: "available" | undefined;
	readonly statusOfSecondary?: "available" | undefined;
}
export default {
	storageAccounts: storageAccounts,
};
