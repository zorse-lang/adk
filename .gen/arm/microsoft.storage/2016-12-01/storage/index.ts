import { ArmResource, Entity as ADKEntity } from "@zorse/adk";

export class storageAccounts
	extends ArmResource<storageAccountsComponentInputs>
	implements storageAccountsComponentOutputs
{
	constructor(entity: ADKEntity, options: storageAccountsComponentInputs) {
		super(entity, options.name, "Microsoft.Storage/storageAccounts", "2016-12-01", options);
	}
	public readonly apiVersion: "2016-12-01";
	public readonly id: string;
	public readonly type: "Microsoft.Storage/storageAccounts";
}
export interface storageAccountsComponentOutputs {
	readonly apiVersion: "2016-12-01";
	readonly id: string;
	readonly type: "Microsoft.Storage/storageAccounts";
}
export interface storageAccountsComponentInputs {
	readonly kind: "BlobStorage";
	readonly location: string;
	readonly name: string;
	readonly properties?: StorageAccountPropertiesCreateParametersOrStorageAccountProperties;
	readonly sku: Sku;
	readonly tags?: StorageAccountCreateParametersTags;
}
export function listAccountSas(resource: storageAccounts, input: AccountSasParameters): ListAccountSasResponse {
	if (resource.apiVersion !== "2016-12-01") {
		throw new Error(`listAccountSas is not available for api version ${resource.apiVersion}`);
	}
	if (resource.type !== "Microsoft.Storage/storageAccounts") {
		throw new Error(`listAccountSas is not available for resource type ${resource.type}`);
	}
	throw new Error("not implemented");
}
export function listKeys(resource: storageAccounts): StorageAccountListKeysResult {
	if (resource.apiVersion !== "2016-12-01") {
		throw new Error(`listKeys is not available for api version ${resource.apiVersion}`);
	}
	if (resource.type !== "Microsoft.Storage/storageAccounts") {
		throw new Error(`listKeys is not available for resource type ${resource.type}`);
	}
	throw new Error("not implemented");
}
export function listServiceSas(resource: storageAccounts, input: ServiceSasParameters): ListServiceSasResponse {
	if (resource.apiVersion !== "2016-12-01") {
		throw new Error(`listServiceSas is not available for api version ${resource.apiVersion}`);
	}
	if (resource.type !== "Microsoft.Storage/storageAccounts") {
		throw new Error(`listServiceSas is not available for resource type ${resource.type}`);
	}
	throw new Error("not implemented");
}
export interface AccountSasParameters {
	readonly keyToSign?: string;
	readonly signedExpiry: string;
	readonly signedIp?: string;
	readonly signedPermission: "a" | "c" | "d" | "l" | "p" | "r" | "u" | "w";
	readonly signedProtocol?: "https";
	readonly signedResourceTypes: "c" | "o" | "s";
	readonly signedServices: "b" | "f" | "q" | "t";
	readonly signedStart?: string;
}
export interface CustomDomain {
	readonly name: string;
	readonly useSubDomainName?: boolean;
}
export interface Encryption {
	readonly keySource: "Microsoft.Storage";
	readonly services?: EncryptionServices;
}
export interface EncryptionService {
	readonly enabled?: boolean;
	readonly lastEnabledTime?: string;
}
export interface EncryptionServices {
	readonly blob?: EncryptionService;
	readonly file?: EncryptionService;
	readonly queue?: EncryptionService;
	readonly table?: EncryptionService;
}
export interface Endpoints {
	readonly blob?: string;
	readonly file?: string;
	readonly queue?: string;
	readonly table?: string;
}
export interface ListAccountSasResponse {
	readonly accountSasToken?: string;
}
export interface ListServiceSasResponse {
	readonly serviceSasToken?: string;
}
export interface ServiceSasParameters {
	readonly canonicalizedResource: string;
	readonly endPk?: string;
	readonly endRk?: string;
	readonly keyToSign?: string;
	readonly rscc?: string;
	readonly rscd?: string;
	readonly rsce?: string;
	readonly rscl?: string;
	readonly rsct?: string;
	readonly signedExpiry?: string;
	readonly signedIdentifier?: string;
	readonly signedIp?: string;
	readonly signedPermission?: "a" | "c" | "d" | "l" | "p" | "r" | "u" | "w";
	readonly signedProtocol?: "https";
	readonly signedResource: "b" | "c" | "f" | "s";
	readonly signedStart?: string;
	readonly startPk?: string;
	readonly startRk?: string;
}
export interface Sku {
	readonly name: "Premium_LRS" | "Standard_GRS" | "Standard_LRS" | "Standard_RAGRS";
	readonly tier?: "Premium";
}
export interface StorageAccountCreateParametersTags {
	readonly [key: string]: string;
}
export interface StorageAccountKey {
	readonly keyName?: string;
	readonly permissions?: "Full";
	readonly value?: string;
}
export interface StorageAccountListKeysResult {
	readonly keys?: StorageAccountKey[];
}
export interface StorageAccountPropertiesCreateParametersOrStorageAccountProperties {
	readonly accessTier?: "Cool";
	readonly creationTime?: string;
	readonly customDomain?: CustomDomain;
	readonly encryption?: Encryption;
	readonly lastGeoFailoverTime?: string;
	readonly primaryEndpoints?: Endpoints;
	readonly primaryLocation?: string;
	readonly provisioningState?: "Creating" | "ResolvingDNS";
	readonly secondaryEndpoints?: Endpoints;
	readonly secondaryLocation?: string;
	readonly statusOfPrimary?: "available";
	readonly statusOfSecondary?: "available";
	readonly supportsHttpsTrafficOnly?: boolean;
}
export default {
	storageAccounts: storageAccounts,
};
