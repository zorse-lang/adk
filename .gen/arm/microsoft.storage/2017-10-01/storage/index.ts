import { ArmResource, Entity as ADKEntity } from "@zorse/adk";

export class storageAccounts
	extends ArmResource<storageAccountsComponentInputs>
	implements storageAccountsComponentOutputs
{
	constructor(entity: ADKEntity, options: storageAccountsComponentInputs) {
		super(entity, options.name, "Microsoft.Storage/storageAccounts", "2017-10-01", options);
	}
	public readonly apiVersion: "2017-10-01";
	public readonly id: string;
	public readonly type: "Microsoft.Storage/storageAccounts";
}
export interface storageAccountsComponentOutputs {
	readonly apiVersion: "2017-10-01";
	readonly id: string;
	readonly type: "Microsoft.Storage/storageAccounts";
}
export interface storageAccountsComponentInputs {
	readonly identity?: Identity | undefined;
	readonly kind: "BlobStorage" | "Storage";
	readonly location: string;
	readonly name: string;
	readonly properties?: StorageAccountPropertiesCreateParametersOrStorageAccountProperties | undefined;
	readonly sku: Sku;
	readonly tags?: StorageAccountCreateParametersTags | undefined;
}
export function listAccountSas(resource: storageAccounts, input: AccountSasParameters): ListAccountSasResponse {
	if (resource.apiVersion !== "2017-10-01") {
		throw new Error(`listAccountSas is not available for api version ${resource.apiVersion}`);
	}
	if (resource.type !== "Microsoft.Storage/storageAccounts") {
		throw new Error(`listAccountSas is not available for resource type ${resource.type}`);
	}
	throw new Error("not implemented");
}
export function listKeys(resource: storageAccounts): StorageAccountListKeysResult {
	if (resource.apiVersion !== "2017-10-01") {
		throw new Error(`listKeys is not available for api version ${resource.apiVersion}`);
	}
	if (resource.type !== "Microsoft.Storage/storageAccounts") {
		throw new Error(`listKeys is not available for resource type ${resource.type}`);
	}
	throw new Error("not implemented");
}
export function listServiceSas(resource: storageAccounts, input: ServiceSasParameters): ListServiceSasResponse {
	if (resource.apiVersion !== "2017-10-01") {
		throw new Error(`listServiceSas is not available for api version ${resource.apiVersion}`);
	}
	if (resource.type !== "Microsoft.Storage/storageAccounts") {
		throw new Error(`listServiceSas is not available for resource type ${resource.type}`);
	}
	throw new Error("not implemented");
}
export interface AccountSasParameters {
	readonly keyToSign?: string | undefined;
	readonly signedExpiry: string;
	readonly signedIp?: string | undefined;
	readonly signedPermission: "a" | "c" | "d" | "l" | "p" | "r" | "u" | "w";
	readonly signedProtocol?: "https" | undefined;
	readonly signedResourceTypes: "c" | "o" | "s";
	readonly signedServices: "b" | "f" | "q" | "t";
	readonly signedStart?: string | undefined;
}
export interface CustomDomain {
	readonly name: string;
	readonly useSubDomainName?: boolean | undefined;
}
export interface Encryption {
	readonly keySource: "Microsoft.Keyvault" | "Microsoft.Storage";
	readonly keyvaultproperties?: KeyVaultProperties | undefined;
	readonly services?: EncryptionServices | undefined;
}
export interface EncryptionService {
	readonly enabled?: boolean | undefined;
	readonly lastEnabledTime?: string | undefined;
}
export interface EncryptionServices {
	readonly blob?: EncryptionService | undefined;
	readonly file?: EncryptionService | undefined;
	readonly queue?: EncryptionService | undefined;
	readonly table?: EncryptionService | undefined;
}
export interface Endpoints {
	readonly blob?: string | undefined;
	readonly file?: string | undefined;
	readonly queue?: string | undefined;
	readonly table?: string | undefined;
}
export interface Identity {
	readonly principalId?: string | undefined;
	readonly tenantId?: string | undefined;
	readonly type: "SystemAssigned";
}
export interface IPRule {
	readonly action?: "Allow" | undefined;
	readonly value: string;
}
export interface KeyVaultProperties {
	readonly keyname?: string | undefined;
	readonly keyvaulturi?: string | undefined;
	readonly keyversion?: string | undefined;
}
export interface ListAccountSasResponse {
	readonly accountSasToken?: string | undefined;
}
export interface ListServiceSasResponse {
	readonly serviceSasToken?: string | undefined;
}
export interface NetworkRuleSet {
	readonly bypass?: ("AzureServices" | "Logging" | "Metrics" | "None") | undefined;
	readonly defaultAction: "Allow";
	readonly ipRules?: IPRule[] | undefined;
	readonly virtualNetworkRules?: VirtualNetworkRule[] | undefined;
}
export interface Restriction {
	readonly reasonCode?: ("NotAvailableForSubscription" | "QuotaId") | undefined;
	readonly type?: string | undefined;
	readonly values?: string[] | undefined;
}
export interface ServiceSasParameters {
	readonly canonicalizedResource: string;
	readonly endPk?: string | undefined;
	readonly endRk?: string | undefined;
	readonly keyToSign?: string | undefined;
	readonly rscc?: string | undefined;
	readonly rscd?: string | undefined;
	readonly rsce?: string | undefined;
	readonly rscl?: string | undefined;
	readonly rsct?: string | undefined;
	readonly signedExpiry?: string | undefined;
	readonly signedIdentifier?: string | undefined;
	readonly signedIp?: string | undefined;
	readonly signedPermission?: ("a" | "c" | "d" | "l" | "p" | "r" | "u" | "w") | undefined;
	readonly signedProtocol?: "https" | undefined;
	readonly signedResource: "b" | "c" | "f" | "s";
	readonly signedStart?: string | undefined;
	readonly startPk?: string | undefined;
	readonly startRk?: string | undefined;
}
export interface Sku {
	readonly capabilities?: SKUCapability[] | undefined;
	readonly kind?: ("BlobStorage" | "Storage") | undefined;
	readonly locations?: string[] | undefined;
	readonly name: "Premium_LRS" | "Standard_GRS" | "Standard_LRS" | "Standard_RAGRS";
	readonly resourceType?: string | undefined;
	readonly restrictions?: Restriction[] | undefined;
	readonly tier?: "Premium" | undefined;
}
export interface SKUCapability {
	readonly name?: string | undefined;
	readonly value?: string | undefined;
}
export interface StorageAccountCreateParametersTags {
	readonly "[ key: string ]"?: string | undefined;
}
export interface StorageAccountKey {
	readonly keyName?: string | undefined;
	readonly permissions?: "Full" | undefined;
	readonly value?: string | undefined;
}
export interface StorageAccountListKeysResult {
	readonly keys?: StorageAccountKey[] | undefined;
}
export interface StorageAccountPropertiesCreateParametersOrStorageAccountProperties {
	readonly accessTier?: "Cool" | undefined;
	readonly creationTime?: string | undefined;
	readonly customDomain?: CustomDomain | undefined;
	readonly encryption?: Encryption | undefined;
	readonly lastGeoFailoverTime?: string | undefined;
	readonly networkAcls?: NetworkRuleSet | undefined;
	readonly primaryEndpoints?: Endpoints | undefined;
	readonly primaryLocation?: string | undefined;
	readonly provisioningState?: ("Creating" | "ResolvingDNS") | undefined;
	readonly secondaryEndpoints?: Endpoints | undefined;
	readonly secondaryLocation?: string | undefined;
	readonly statusOfPrimary?: "available" | undefined;
	readonly statusOfSecondary?: "available" | undefined;
	readonly supportsHttpsTrafficOnly?: boolean | undefined;
}
export interface VirtualNetworkRule {
	readonly action?: "Allow" | undefined;
	readonly id: string;
	readonly state?: ("deprovisioning" | "failed" | "networkSourceDeleted" | "provisioning") | undefined;
}
export default {
	storageAccounts: storageAccounts,
};
