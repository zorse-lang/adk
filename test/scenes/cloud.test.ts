import * as ECS from "@zorse/adk/core/object";

import { ArmScene, AwsScene, GdmScene, RosScene } from "@zorse/adk/scenes";

import { storageAccounts as ARMBucket } from "@zorse/adk/clouds/arm/microsoft.storage/2022-09-01/storage";
import { Bucket as AWSBucket } from "@zorse/adk/clouds/cfn/aws.s3/2010-09-09";
import { Buckets as GDMBucket } from "@zorse/adk/clouds/gdm/gcloud.storage/v1/buckets";
import { Bucket as ROSBucket } from "@zorse/adk/clouds/ros/aliyun.oss/2015-09-01";

describe("Cloud Scene tests", () => {
	it("should be able to compose basic CFN templates", async () => {
		const system = new ECS.System();
		new AwsScene(system);
		const entity = new ECS.Entity(system);
		const bucket1 = new AWSBucket(entity, { LogicalId: "Bucket1" });
		new AWSBucket(entity, { LogicalId: "Bucket2", BucketName: bucket1.Arn });
		const composition = await system.compose();
		const result = composition.gizmos();
		expect(result).toMatchSnapshot();
	});

	it("should be able to compose basic ROS templates", async () => {
		const system = new ECS.System();
		new RosScene(system);
		const entity = new ECS.Entity(system);
		const bucket1 = new ROSBucket(entity, { LogicalId: "Bucket1", BucketName: "test-bucket1" });
		new ROSBucket(entity, { LogicalId: "Bucket2", BucketName: bucket1.DomainName });
		const composition = await system.compose();
		const result = composition.gizmos();
		expect(result).toMatchSnapshot();
	});

	it("should be able to compose basic GDM templates", async () => {
		const system = new ECS.System();
		new GdmScene(system);
		const entity = new ECS.Entity(system);
		const bucket1 = new GDMBucket(entity, {
			name: "Bucket1",
			storageClass: "STANDARD",
			location: "US",
		});
		new GDMBucket(entity, {
			name: "Bucket2",
			kind: bucket1.properties.kind,
		});
		const composition = await system.compose();
		const result = composition.gizmos();
		expect(result).toMatchSnapshot();
	});

	it("should be able to compose basic ARM templates", async () => {
		const system = new ECS.System();
		new ArmScene(system);
		const entity = new ECS.Entity(system);
		const bucket1 = new ARMBucket(entity, {
			name: "Bucket1",
			location: "westus",
			sku: { name: "Standard_LRS" },
			kind: "BlobStorage",
		});
		new ARMBucket(entity, {
			name: "Bucket2",
			location: "westus",
			sku: { name: "Standard_LRS" },
			kind: "BlobStorage",
			tags: { "test-tag": `tag-${bucket1.id}` },
		});
		const composition = await system.compose();
		const result = composition.gizmos();
		expect(result).toMatchSnapshot();
	});
});
