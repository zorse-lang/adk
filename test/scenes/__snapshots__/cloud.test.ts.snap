// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cloud Scene tests should be able to compose basic ARM templates 1`] = `
[
  {
    "Components": [
      {
        "apiVersion": "2022-09-01",
        "name": "Bucket1",
        "properties": {
          "kind": "BlobStorage",
          "location": "westus",
          "name": "Bucket1",
          "sku": {
            "name": "Standard_LRS",
          },
        },
        "type": "Microsoft.Storage/storageAccounts",
      },
      {
        "apiVersion": "2022-09-01",
        "name": "Bucket2",
        "properties": {
          "kind": "BlobStorage",
          "location": "westus",
          "name": "Bucket2",
          "sku": {
            "name": "Standard_LRS",
          },
          "tags": {
            "test-tag": "tag-[reference(resourceId('Microsoft.Storage/storageAccounts', 'Bucket1')).id]",
          },
        },
        "type": "Microsoft.Storage/storageAccounts",
      },
    ],
  },
]
`;

exports[`Cloud Scene tests should be able to compose basic CFN templates 1`] = `
[
  {
    "Components": [
      {
        "Bucket1": {
          "Properties": {
            "LogicalId": "Bucket1",
          },
          "Type": "AWS::S3::Bucket",
        },
      },
      {
        "Bucket2": {
          "Properties": {
            "BucketName": {
              "Fn::GetAtt": [
                "Bucket1",
                "Arn",
              ],
            },
            "LogicalId": "Bucket2",
          },
          "Type": "AWS::S3::Bucket",
        },
      },
    ],
  },
]
`;

exports[`Cloud Scene tests should be able to compose basic GDM templates 1`] = `
[
  {
    "Components": [
      {
        "name": "Bucket1",
        "properties": {
          "location": "US",
          "name": "Bucket1",
          "storageClass": "STANDARD",
        },
        "type": "storage.v1.Buckets",
      },
      {
        "name": "Bucket2",
        "properties": {
          "kind": "$(ref.Bucket1.properties.kind)",
          "name": "Bucket2",
        },
        "type": "storage.v1.Buckets",
      },
    ],
  },
]
`;

exports[`Cloud Scene tests should be able to compose basic ROS templates 1`] = `
[
  {
    "Components": [
      {
        "Bucket1": {
          "Properties": {
            "BucketName": "test-bucket1",
            "LogicalId": "Bucket1",
          },
          "Type": "ALIYUN::OSS::Bucket",
        },
      },
      {
        "Bucket2": {
          "Properties": {
            "BucketName": {
              "Fn::GetAtt": [
                "Bucket1",
                "DomainName",
              ],
            },
            "LogicalId": "Bucket2",
          },
          "Type": "ALIYUN::OSS::Bucket",
        },
      },
    ],
  },
]
`;
